txcache/baseTxCache.go

```diff
@@ -0,0 +1,44 @@
+package txcache
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-storage-go/common"
+)
+
+const maxNumOfEvictionWorkers = 5
+
+type evictionWorkerPool interface {
+	Stop()
+	Submit(task func())
+}
+
+type baseTxCache struct {
+	mutEvictionHandlers sync.RWMutex
+	evictionHandlers    []func(txHash []byte)
+	evictionWorkerPool  evictionWorkerPool
+}
+
+// RegisterEvictionHandler registers a handler which will be called when a tx is evicted from cache
+func (cache *baseTxCache) RegisterEvictionHandler(handler func(hash []byte)) error {
+	if handler == nil {
+		return common.ErrNilEvictionHandler
+	}
+
+	cache.mutEvictionHandlers.Lock()
+	cache.evictionHandlers = append(cache.evictionHandlers, handler)
+	cache.mutEvictionHandlers.Unlock()
+
+	return nil
+}
+
+// notifyEvictionHandlers will be called on a separate go routine
+func (cache *baseTxCache) notifyEvictionHandlers(txHashes [][]byte) {
+	cache.mutEvictionHandlers.RLock()
```

Comment:
pushed

Reply:
I think you need to do a copy() as otherwise you access the same backing array and you will still get race conditions

Also I would do a go handler(), I don't think it needs to be sequential

Reply:
moved the worker pool on notifications